lib/native/source/opensmt/api.patch

```diff
@@ -0,0 +1,537 @@
+ 
++    PTRef parseFormula(std::string input);
++
+     PTRef instantiateFunctionTemplate(TemplateFunction const & tmplt, vec<PTRef> const & args);
+ 
+     SSymRef       getSortSymIndexed()              const { return sym_IndexedSort; }
+diff --git a/src/logics/LogicFactory.cc b/src/logics/LogicFactory.cc
+index ceedd296..7e07010d 100644
+--- a/src/logics/LogicFactory.cc
++++ b/src/logics/LogicFactory.cc
+@@ -79,7 +79,9 @@ Logic * opensmt::LogicFactory::getInstance(Logic_t logicType) {
+     return l;
+ }
+ 
+-ArithLogic * opensmt::LogicFactory::getLRAInstance() { return new ArithLogic(Logic_t::QF_LRA); }
++ArithLogic * opensmt::LogicFactory::getLAInstance(Logic_t logicType) { return dynamic_cast<ArithLogic*>(getInstance(logicType)); }
+ 
++ArithLogic * opensmt::LogicFactory::getLRAInstance() { return new ArithLogic(Logic_t::QF_LRA); }
+ ArithLogic * opensmt::LogicFactory::getLIAInstance() { return new ArithLogic(Logic_t::QF_LIA); }
+ 
++ArithLogic * opensmt::LogicFactory::getLogicAll() { return new ArithLogic(Logic_t::QF_AUFLIRA); }
+diff --git a/src/logics/LogicFactory.h b/src/logics/LogicFactory.h
+index 65b039e9..120446d3 100644
+--- a/src/logics/LogicFactory.h
++++ b/src/logics/LogicFactory.h
+@@ -169,9 +169,13 @@ std::string getStringFromLogic(const Logic_t logic);
+ 
+ class LogicFactory {
+ public:
+-    static Logic * getInstance(Logic_t);
++    static Logic *  getInstance(Logic_t);
++    static ArithLogic * getLAInstance(Logic_t);
++
+     static ArithLogic * getLRAInstance();
+     static ArithLogic * getLIAInstance();
++
++    static ArithLogic * getLogicAll();
+ };
+ }
+ #endif //OPENSMT_LOGICFACTORY_H
+diff --git a/src/logics/Theory.h b/src/logics/Theory.h
+index ef231407..0799d499 100644
+--- a/src/logics/Theory.h
++++ b/src/logics/Theory.h
+@@ -102,7 +102,7 @@ struct PFRef {
+     inline friend bool operator== (const PFRef& a1, const PFRef& a2) { return a1.x == a2.x; };
+ };
+ 
+-const static struct PFRef PFRef_Undef = {INT32_MAX};
++static const struct PFRef PFRef_Undef = {INT32_MAX};
+ 
+ // No global variable for storing the push frame id.  I decided to
+ // implement this as my own memory allocation.  There's no free though
+diff --git a/src/models/Model.cc b/src/models/Model.cc
+index 1ab10b07..2f182d05 100644
+--- a/src/models/Model.cc
++++ b/src/models/Model.cc
+@@ -114,4 +114,5 @@ TemplateFunction Model::getDefinition(SymRef sr) const {
+         }
+         return TemplateFunction(symName, formalArgs, logic.getSym(sr).rsort(), logic.getDefaultValuePTRef(logic.getSym(sr).rsort()));
+     }
+-}
+\ No newline at end of file
++
++}
+diff --git a/src/sorts/SSort.h b/src/sorts/SSort.h
+index 990da3b0..14922397 100644
+--- a/src/sorts/SSort.h
++++ b/src/sorts/SSort.h
+@@ -40,7 +40,7 @@ struct SSymRef {
+     inline friend bool operator!= (SSymRef a1, SSymRef a2) {return a1.x != a2.x; }
+ };
+ 
+-const static struct SSymRef SSymRef_Undef = {INT32_MAX};
++static const struct SSymRef SSymRef_Undef = {INT32_MAX};
```

Comment:
Why do we change the ordering of modifiers here?

Reply:
This seems to happen just to avoid compilation problems later in the prozess, because "static" should come first. Oh no ;-)

Reply:
Yes, SWIG can't parse the header and simply crashes if the order of the modifies is not changed.

Reply:
`const static` still seems to be legal C++, but it looks like `static const` is much more preferred. We can fix this.